src/libtiled/wangset.cpp

```diff
@@ -470,6 +521,42 @@ bool WangSet::wangIdIsUsed(WangId wangId) const
     return mWangIdToWangTile.contains(wangId);
 }
 
+bool WangSet::wildWangIdIsUsed(WangId wangId) const
+{
+    if (isEmpty())
+        return false;
+    if (!wangId)
+        return true;
+
+    auto i = wangId.variationsBegin(mEdgeColors, mCornerColors);
+    auto end = wangId.variationsEnd(mEdgeColors, mCornerColors);
+
+    while (true) {
+        if (wangIdIsUsed(*i))
+            return true;
+
+        if (++i == end) {
+            if (wangIdIsUsed(*i))
+                return true;
+            break;
+        }
+    }
+
+    return false;
+}
+
+bool WangSet::isComplete() const
+{
+    if (isEmpty())
+        return false;
+    return (unsigned) mWangIdToWangTile.uniqueKeys().size() == completeSetSize();
+}
+
+unsigned WangSet::completeSetSize() const
+{
+    return qPow(mEdgeColors, 4) * qPow(mCornerColors, 4);
```

Comment:
> We can't go out of int range here anyway due to limitations on the color counts, right?

I think we can. Max color is 15, so max return is 15^8 = 2562890625 > 0x7fffffff

To avoid casting, I the new function for unique keys will return unsigned as well.